src/Leaflet/Halogen.purs

```diff
@@ -0,0 +1,153 @@
+module Leaflet.Halogen where
+
+import Prelude
+
+import Control.Monad.Eff (kind Effect)
+import Control.Monad.Aff (delay)
+import Control.Monad.Aff.Class (class MonadAff, liftAff)
+
+import CSS.Geometry (width, height)
+import CSS.Size (px, pct)
+
+import Data.Int as Int
+import Data.Maybe (Maybe(..), fromJust)
+import Data.Traversable as F
+import Data.Time.Duration (Milliseconds(..))
+import Data.Tuple (Tuple(..), fst)
+
+import DOM (DOM)
+
+import Halogen as H
+import Halogen.HTML as HH
+import Halogen.HTML.CSS (style)
+import Halogen.HTML.Properties as HP
+
+import Leaflet.Core as LC
+import Leaflet.Util ((∘))
+
+import Partial.Unsafe (unsafePartial)
+
+type State =
+  { width ∷ Int
+  , height ∷ Int
+  , view ∷ LC.LatLng
+  , zoom ∷ LC.Zoom
+  , leaflet ∷ Maybe LC.Leaflet
+  }
+
+data Query a
+  = Init a
+  | SetDimension { width ∷ Maybe Int, height ∷ Maybe Int } a
+    -- Note, layers are not preserved in state, parent component must
+    -- handle it.
+  | AddLayers (Array LC.Layer) a
+  | RemoveLayers (Array LC.Layer) a
+  | GetLeaflet (Maybe LC.Leaflet → a)
+  | SetView LC.LatLng a
+  | SetZoom LC.Zoom a
+
+data Message
+  = Initialized LC.Leaflet
+
+type Effects (e ∷ # Effect) =
+  ( dom ∷ DOM
+  | e )
+
+type HTML = H.ComponentHTML Query
+
+type DSL m = H.ComponentDSL State Query Message m
+
+type Dimensions = { width ∷ Int, height ∷ Int }
+
+initialState ∷ Dimensions → State
+initialState {width, height} =
+  { height
+  , width
+  , view: unsafePartial fromJust $ LC.mkLatLng (-37.87) 175.457
+  , zoom: unsafePartial fromJust $ LC.mkZoom 12
+  , leaflet: Nothing
+  }
+
+leaflet
+  ∷ ∀ e m
+  . MonadAff (Effects e) m
+  ⇒ H.Component HH.HTML Query (Tuple Dimensions Unit) Message m
+leaflet = leaflet' $ fst >>> \dim →
+  Just $ H.action $ SetDimension { width : Just dim.width, height: Just dim.height }
+
+
+leaflet'
+  ∷ ∀ e m i
+  . MonadAff (Effects e) m
+  ⇒ (Tuple Dimensions i → Maybe (Query Unit))
+  → H.Component HH.HTML Query (Tuple Dimensions i) Message m

```

Comment:
Well... I can imagine situation where input is `Tuple Dimensions (Maybe Zoom)` and in if `Just zoom` receiver sends query `SetZoom zoom`. Or, since this component don't manage child layers, it can be really useful to have array of layers as input type. Anyways, I don't think that width/height pair is special here and I'd prefer or remove input completely or make it extensible (at least as second value in tuple).